Sources/PackageModel/PackageReference.swift

```diff
@@ -29,46 +27,19 @@ public struct PackageReference: Codable {
         case remote
     }
 
-    /// Compute identity of a package given its URL.
-    public static func computeIdentity(packageURL: String) -> String {
-        return computeDefaultName(fromURL: packageURL).lowercased()
-    }
-
-    /// Compute the default name of a package given its URL.
-    public static func computeDefaultName(fromURL url: String) -> String {
-      #if os(Windows)
-        let isSeparator : (Character) -> Bool = { $0 == "/" || $0 == "\\" }
-      #else
-        let isSeparator : (Character) -> Bool = { $0 == "/" }
-      #endif
-
-        // Get the last path component of the URL.
-        // Drop the last character in case it's a trailing slash.
-        var endIndex = url.endIndex
-        if let lastCharacter = url.last, isSeparator(lastCharacter) {
-            endIndex = url.index(before: endIndex)
-        }
-
-        let separatorIndex = url[..<endIndex].lastIndex(where: isSeparator)
-        let startIndex = separatorIndex.map { url.index(after: $0) } ?? url.startIndex
-        var lastComponent = url[startIndex..<endIndex]
-
-        // Strip `.git` suffix if present.
-        if lastComponent.hasSuffix(".git") {
-            lastComponent = lastComponent.dropLast(4)
-        }
-
-        return String(lastComponent)
+    /// The identity of the package.
+    public var identity: String {
```

Comment:
I'd love to have this expose the "real" identity instead of hiding it. we can then add a function/var to return the computed name projection and for the call sites that *absolutely must*  use that

Reply:
My plan was to do that in the next step, when we do make the semantic change. That way, the API `identity` remains consistent with the actual behavior of the system throughout the process, and we avoid changing the interface and the implementation both at the same time.

But if you'd prefer to do this now, we could certainly rename this symbol and add a new `identity` property that returns a `PackageIdentity` value. All I'd need to know is what that would be called. Best I can come up with are `oldIdentity` and `legacyIdentity`, which both have a code smell.

Reply:
I would prefer to carry PackageIdentity anywhere we have identity over String. @neonichu @abertelrud ?

Reply:
I agree — it would be good (especially to make sure we don't have to re-audit) to switch to a specific value type here rather than String, and it would be also be a good place to attach type documentation that summarizes the semantic intent of identity (not necessarily details about how the underlying string is formed).

Reply:
With c3fbf4e08ddd54a24fd24b0900702b3e747ad65a, we have an explicit typealias (`PackageReference.Identity`) and property (`PackageIdentity.legacyIdentity`). When it comes time change the semantics, we can audit for those APIs explicitly.